specs/accordion.md

```diff
@@ -0,0 +1,173 @@
+
+### Prior Art/Examples
+
+- [Ant Design](https://ant.design/components/collapse/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/accordion/usage/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/accordion/)
+- [Fluent UI](https://fluentsite.z22.web.core.windows.net/components/accordion/accessibility)
+- [Primer](https://primer.style/components/Details)
+- [Bootstrap](https://getbootstrap.com/docs/4.3/components/collapse/)
+
+---
+
+### API
+
+**Accordion**
+*Component Name*
+- `fast-accordion`
+
+*Attributes:*
+- `expand` - enum
+  - single - default
+  - multi
+
+*Events*
+- `change: CustomEvent`
+  - no custom data
+  - bubbles
+
+**Accordion Header**
+*Component names:*
+- `fast-accordion-header`
+
+*Attributes:*
+- `id` - string
+
+*Parts:*
+- button
+
+*Slot Names*
+- expanded-glyph
+- collapsed-glyph
+
+**Accordion Panel**
+*Component names:*
+- `fast-accordion-panel`
+
+*Attributes:*
+- `id` - string
+
+### Anatomy and Appearance
+
+```HTML
+<!-- shadow root -->
+<div>
+    <h3>
+        <button
+            type="button"
+            part="button"
+            aria-expanded="true"
+            class="button"
+            aria-controls="panel1"
+            id="accordion1"
+        >
+            Panel one
+            <slot name="expanded-glyph" part="expanded-glyph"></slot>
+            <slot name="collapsed-glyph" part="collapsed-glyph"></slot>
+       </button>
+    </h3>
+    <div
+        id="panel1"
+        role="region"
+        aria-labelledby="accordion1"
+    >
+        Panel one content
+    </div>
+    <h3>
+        <button
+            type="button"
+            aria-expanded="false"
+            class="button"
+            aria-controls="panel2"
+            id="accordion2"
+        >
+            Panel one
+            <slot name="expanded-glyph" part="expanded-glyph"></slot>
+            <slot name="collapsed-glyph" part="collapsed-glyph"></slot>
+       </button>
+    </h3>
+    <div
+        id="panel2"
+        part="panel"
+        role="region"
+        aria-labelledby="accordion2"
+        class="panel"
+    >
+        Panel one content
+    </div>
+    <h3>
+        <button
+            type="button"
+            aria-expanded="true"
+            class="button"
+            aria-controls="panel3"
+            id="accordion3"
+        >
+            Panel one
+            <slot name="expanded-glyph" part="expanded-glyph"></slot>
+            <slot name="collapsed-glyph" part="collapsed-glyph"></slot>
+       </button>
+    </h3>
+    <div
+        id="panel3"
+        part="panel"
+        role="region"
+        aria-labelledby="accordion3"
+        class="panel"
+    >
+        Panel one content
+    </div>
+</div>
+
+<!-- end shadow root -->
+```
+
+---
+
+## Implementation
+
+```HTML
+<fast-accordion>
+    <fast-accordion-header>Panel one</fast-accordion-header>
```

Comment:
Did you consider a template structure where the header is a slot of the panel?

```
<fast-accordion>
   <fast-accordion-panel>
      <h3 slot="header">panel header</h3>
      <p>I'm some content</p>
    </fast-accordion-panel>
</fast-accordion>

Reply:
Yeah, I think we have to do something like this. I realized that having the `header` and `panel` separated would not be possible in order to get the alternating stacked DOM structure we need. For example:

```
<header/>
<panel/>
<header/>
<panel/>
<header/>
<panel/>
```

Reply:
I think we will still need `<fast-accordion-header/>` which will be inside the something like `<fast-accordion-item/>` like so:

```HTML
<fast-accordion>
    <fast-accordion-item>
        <fast-accordion-header headingtag="h2">Header One</fast-accordion-header>
        Panel one Content
    </fast-accordion-item>
    <fast-accordion-item expanded>
        <fast-accordion-header headingtag="h2" expanded>Header Two</fast-accordion-header>
        Panel two content
    </fast-accordion-item>
    <fast-accordion-item>
        <fast-accordion-header headingtag="h2">Header Three</fast-accordion-header>
        Panel three content
    </fast-accordion-item>
</fast-accordion>
```